pkg/gen/node-templates/provider.ts.mustache

```diff
@@ -18,19 +18,23 @@ export namespace {{Group}} {
 
       {{/Properties}}
 
+      public getInputs(): inputApi.{{Group}}.{{Version}}.{{Kind}} { return this.__inputs; }
+      private readonly __inputs: inputApi.{{Group}}.{{Version}}.{{Kind}};
+
       /**
       * Create a {{Group}}.{{Version}}.{{Kind}} resource with the given unique name, arguments, and options.
       *
       * @param name The _unique_ name of the resource.
       * @param args The arguments to use to populate this resource's properties.
       * @param opts A bag of options that control this resource's behavior.
       */
-      constructor(name: string, args?: inputApi.{{Group}}.{{Version}}.{{Kind}}, opts?: pulumi.ResourceOptions) {
+      constructor(name: string, args: inputApi.{{Group}}.{{Version}}.{{Kind}}, opts?: pulumi.ResourceOptions) {
           let inputs: pulumi.Inputs = {};
           {{#Properties}}
           inputs["{{Name}}"] = {{{DefaultValue}}};
           {{/Properties}}
           super("kubernetes:{{APIVersion}}:{{Kind}}", name, inputs, opts);
+          this.__inputs = args;
```

Comment:
Should this be `args` or `inputs`? Or are we expecting that they'll always be equal at this point?

Reply:
Resolved offline. `this.__inputs` is different from `inputs`, so we'll be keeping this code as-is.